origin-js/src/contractInterface/users/v00_adapter.js

```diff
@@ -183,15 +190,31 @@ class V00_UsersAdapter {
     if (profileClaims.length) {
       const bytes32 = profileClaims[profileClaims.length - 1].data
       const ipfsHash = this.contractService.getIpfsHashFromBytes32(bytes32)
-      profile = await this.ipfsDataStore.load(PROFILE_DATA_TYPE, ipfsHash)
+      try {
+        profile = await this.ipfsDataStore.load(PROFILE_DATA_TYPE, ipfsHash)
+      } catch (error) {
+        console.error(`Can not read profile data from ipfs (hash: '${ipfsHash}'): ${error.message}`)
+      }
     }
+
     const validAttestations = await this.validAttestations(
       identityAddress,
       nonProfileClaims
     )
-    const attestations = validAttestations.map(
-      att => new AttestationObject(att)
-    )
+
+    const attestations = validAttestations
+      .map(att => {
+        try {
+          if (ClaimDataIsIpfsHash.includes(att.topic))
+            att.ipfsHash = this.contractService.getIpfsHashFromBytes32(att.data)
+          return att
+        }
+        catch (error) {
+          console.error(`Can not convert to ipfs hash: ${error.message}`)
```

Comment:
Would be nice if we would keep backward compatibility and still support legacy Airbnb/Twitter attestations that were not storing data in IPFS ?

Reply:
good point. Have changed it that it is always a valid attestation. 

There is one downside though, when testing with old attestations, this function `this.contractService.getIpfsHashFromBytes32` will most of the time succeed and come up with something that looks like an IPFS hash but it is not. We have previously populated the data field on the bridge server and hashed it in the DApp before storing it. Twitter attestation used to have non changing data (string "twitter verified" ) but Airbnb had an id in the data field (https://github.com/OriginProtocol/origin/blob/master/origin-bridge/logic/attestation_service.py#L464). Meaning after hashing it in the DApp we can not recognise legacy data attestation with the current data attestations (for Twitter we can). 

With the current implementation the DApp needs to expect that not all ipfs fileds in attestations are valid. Some are legacy that do not resove to a file in IPFS (even though they will appear to have an IPFS hash location) @micahalcorn 

Can anyone think of a better solution for this? Should we make a special case to recognise legacy Twitter attestations?

Reply:
It is hacky but could we look at the block number the attestation was issued at (I think the event should include the blockNumber) and have a cut-off hard coded constant that indicates if it is old style attestation vs new style ? e.g. any attestation with blockNumber > cutoff is new style

Reply:
That is a good idea. Have added a new property in `origin-dapp.env` file. The only caveat is when deploying for the first time on `Rinkeby` and `Mainnet` that variable needs to be set. The current block height can be fetched in browser in console: 
```
await window.web3.eth.getBlockNumber()
```